module/zfs/abd.c

```diff
@@ -1050,6 +1050,29 @@ abd_cmp(abd_t *dabd, abd_t *sabd)
 	    abd_cmp_cb, NULL));
 }
 
+/*
+ * Check if ABD content is all-zeroes.
+ */
+static int
+abd_cmp_zero_off_cb(void *data, size_t len, void *private)
+{
+	(void) private;
+
+	uint64_t *end = (uint64_t *)((char *)data + len);
+	for (uint64_t *word = (uint64_t *)data; word < end; word++)
+		if (*word != 0)
+			return (1);
+
+	return (0);
+}
+
+int
+abd_cmp_zero_off(abd_t *abd, size_t off, size_t size)
+{
+	return (abd_iterate_func(abd, off, size, abd_cmp_zero_off_cb, NULL)
+	    ?  1 : 0);
```

Comment:
The first thing that comes to mind when seeing `? 1 : 0` is "redundant". Maybe use `!!` if returning `0/1` is strictly required?

Reply:
`!!` is a linux'ism, while functionally it is just the same, I suppose even down to assembler.  I was thinking whether `boolean_t` there would be more reasonable than int, but I suppose int was copied from existing functions there.

Reply:
Oh good get, just leftovers from an earlier incarnation.

And yes, it's a `adb_cmp_*` just to make it look like the others. It was `boolean_t abd_is_zero(...)` but to me it felt like it should have some relationship with `abd_zero()` and I didn't want any confusion.